be/src/io/fs/benchmark/s3_benchmark.hpp

```diff
@@ -52,4 +55,164 @@ class S3ReadBenchmark : public BaseBenchmark {
+            : BaseBenchmark("S3SizeBenchmark", iterations, conf_map) {}
+    virtual ~S3SizeBenchmark() = default;
+
+    Status init() override {
+        bm_log("begin to init {}", _name);
+        std::string file_path = _conf_map["file"];
+        io::FileReaderOptions reader_opts = FileFactory::get_reader_options(nullptr);
+        RETURN_IF_ERROR(
+                FileFactory::create_s3_reader(_conf_map, file_path, &_fs, &_reader, reader_opts));
+        bm_log("finish to init {}", _name);
+        return Status::OK();
+    }
+
+    Status run() override {
+        size_t i = _reader->size();
+        bm_log("size result is: {}", i);
+        return Status::OK();
+    }
+
+private:
+    std::shared_ptr<io::FileSystem> _fs;
+    io::FileReaderSPtr _reader;
+};
+
+class S3ListBenchmark : public BaseBenchmark {
+public:
+    S3ListBenchmark(int iterations, const std::map<std::string, std::string>& conf_map)
+            : BaseBenchmark("S3ListBenchmark", iterations, conf_map) {}
+    virtual ~S3ListBenchmark() = default;
+
+    Status init() override {
+        bm_log("begin to init {}", _name);
+        S3URI s3_uri(_conf_map["file"]);
+        RETURN_IF_ERROR(s3_uri.parse());
+        RETURN_IF_ERROR(
+                S3ClientFactory::convert_properties_to_s3_conf(_conf_map, s3_uri, &_s3_conf));
+        RETURN_IF_ERROR(io::S3FileSystem::create(std::move(_s3_conf), "", &_fs));
+        bm_log("finish to init {}", _name);
+        return Status::OK();
+    }
+
+    Status run() override {
+        std::vector<FileInfo> files;
+        bool exists = true;
+        _fs->list(_conf_map["file"], true, &files, &exists);
+        bm_log("list result is: {}", files.size());
+        return Status::OK();
+    }
+
+private:
+    doris::S3Conf _s3_conf;
+    std::shared_ptr<io::S3FileSystem> _fs;
+};
+
+class S3OpenBenchmark : public BaseBenchmark {
+public:
+    S3OpenBenchmark(int iterations, const std::map<std::string, std::string>& conf_map)
+            : BaseBenchmark("S3OpenBenchmark", iterations, conf_map) {}
+    virtual ~S3OpenBenchmark() = default;
+
+    Status init() override {
+        bm_log("begin to init {}", _name);
+        S3URI s3_uri(_conf_map["file"]);
+        RETURN_IF_ERROR(s3_uri.parse());
+        RETURN_IF_ERROR(
+                S3ClientFactory::convert_properties_to_s3_conf(_conf_map, s3_uri, &_s3_conf));
+        RETURN_IF_ERROR(io::S3FileSystem::create(std::move(_s3_conf), "", &_fs));
+
+        std::string file_path = _conf_map["file"];
+        io::FileReaderOptions reader_opts = FileFactory::get_reader_options(nullptr);
+        RETURN_IF_ERROR(FileFactory::create_s3_reader(
+                _conf_map, file_path, reinterpret_cast<std::shared_ptr<io::FileSystem>*>(&_fs),
+                &_reader, reader_opts));
+        bm_log("finish to init {}", _name);
+        return Status::OK();
+    }
+
+    Status run() override { return _fs->open_file(_conf_map["file"], &_reader); }
```

Comment:
Missing `const FileReaderOptions& reader_options` in `open_file`'s parameter list?